src/cpl.rs

```diff
@@ -125,7 +228,121 @@ impl Clone for CslStringList {
     }
 }
 
+impl Debug for CslStringList {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut b = f.debug_tuple("CslStringList");
+
+        for e in self.iter() {
+            b.field(&e.to_string());
+        }
+
+        b.finish()
+    }
+}
+
+impl Display for CslStringList {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        // CSLPrint would be preferable here, but it can only write to a file descriptor.
+        let len = self.len();
+        for i in 0..len {
+            if let Some(field) = self.get_field(i) {
+                f.write_fmt(format_args!("{field}\n"))?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl PartialEq for CslStringList {
+    fn eq(&self, other: &Self) -> bool {
+        // Not the best comparison mechanism, but C API doesn't help us here.
+        self.to_string() == other.to_string()
+    }
+}
+
+/// Parse a space-delimited string into a [`CslStringList`].
+///
+/// See [`CSLTokenizeString`](https://gdal.org/api/cpl.html#_CPPv417CSLTokenizeStringPKc) for details
+impl FromStr for CslStringList {
+    type Err = GdalError;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        // See: https://github.com/OSGeo/gdal/blob/cd2a054b0d7b881534baece69a8f52ddb69a53d9/port/cpl_string.h#L86C1-L97
+        static CSLT_HONOURSTRINGS: c_int = 0x0001;
+        static CSLT_PRESERVEESCAPES: c_int = 0x0008;
+        static DELIM: &[u8; 4] = b" \n\t\0";
+
+        let cstr = CString::new(s)?;
+        let c_list = unsafe {
+            CSLTokenizeString2(
+                cstr.as_ptr(),
+                DELIM.as_ptr() as *const c_char,
+                CSLT_HONOURSTRINGS | CSLT_PRESERVEESCAPES,
+            )
+        };
+        Ok(Self { list_ptr: c_list })
+    }
+}
+
+/// Convenience for creating a [`CslStringList`] from a slice of _key_/_value_ tuples.
```

Comment:
It's a little annoying that this is quadratic. Can you add a comment mentioning that?

Reply:
I was wondering if we should get rid of it all together... Adding `FromStr` makes it much less necessary.

Can you think of any way to _not_ make it quadratic? Seems like the whole API is ironically inefficient.

Reply:
Nope, I don't think we can. I like kinda' like the API more than `FromStr`, though.

Reply:
Unless you have objections, I'm going to remove it.

Reply:
It's useful when you don't know the values at compile-time. You'd need to format a string otherwise.

Reply:
Another problem with it is that I couldn't get it to work with generic `&[(&str, &str)]` slices. IOW, it _does_ only work if you know the length at compile time:

```rust
impl<const N: usize> TryFrom<&[(&str, &str); N]> for CslStringList
```

If you remove the `const` parameter, you get the following compiler error:

<details><summary>Details</summary>
<pre>
error[E0277]: the trait bound `cpl::CslStringList: From<&[(&str, &str); 2]>` is not satisfied
   --> src/cpl.rs:518:41
    |
518 |         let l = CslStringList::try_from(&[("ONE", "1"), ("TWO", "2")])?;
    |                 ----------------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<&[(&str, &str); 2]>` is not implemented for `cpl::CslStringList`
    |                 |
    |                 required by a bound introduced by this call
    |
    = help: the following other types implement trait `TryFrom<T>`:
              <cpl::CslStringList as TryFrom<RasterizeOptions>>
              <cpl::CslStringList as TryFrom<&[(&str, &str)]>>
    = note: required for `&[(&str, &str); 2]` to implement `Into<cpl::CslStringList>`
    = note: required for `cpl::CslStringList` to implement `TryFrom<&[(&str, &str); 2]>`
</pre>
</details>

Reply:
I'm going to decline adding a comment right now. Not quite sure how to phrase it given it's a problem with the C API and not the Rust bindings, and also I'm not happy with the current implementation at all.

Reply:
I think that was just me trying to be cute about this, so `try_from(["ONE", "1"])` would work without the `&`. But looks like it doesn't work anyway.

We can replace that impl with `impl TryFrom<&[(&str, &str)]> for CslStringList`, which should work in your example.

Reply:
As for the comment, we can just say "Due to the GDAL API, this is quadratic and might be slow for large number of entries. Consider using `FromStr` instead.".

Reply:
@lnicola I'll revisit this in my next PR. I have to add a `merge` method anyway.

Reply:
Ah, that might need an `as_slice` on the caller side. Anyway, we should probably implement `FromIterator` and `Extend`? I'm not sure.

Reply:
`FromIterator` is a great idea. I tried parameterizing on `AsRef<[(&str, &str)]>` but got into orphan rule issues.